Makefile

```diff
@@ -32,3 +32,8 @@ phpstan:
 .PHONY: phpstan-generate-baseline
 phpstan-generate-baseline:
 	php vendor/bin/phpstan analyse -l 8 -c phpstan.neon src tests -b phpstan-baseline.neon
+
+.PHONY: infection
+infection:
+	composer require --dev infection/infection -W --ignore-platform-req=ext-mongodb
+	vendor/bin/infection --logger-github=false
```

Comment:
Why turning off the logger-github?

Reply:
Also personally I wouldn't add this to `Makefile`. We'll need a more complex logic in the GHA job.

Reply:
> mutation job on php 8.3 might error about a mutation which can only be killed in the php 8.1 test-suite

I had this concern as well but I talked to Maks about this and given that Infection would only mutate covered lines then this shouldn't happen. If a test ran on PHP 8.3 covers a line and we mutate that line, it should fail the test on 8.3.

Reply:
If this was really true Infection would be unusable for us. We'd have to be able to invoke each phase separately and modify the results before feeding them to the next phase. (Run tests on each PHP version, feed the data into Infection, let it mutate the sources, run tests on each PHP version again ourselves and so on.)

Reply:
I think we could run infection on multiple php versions in parallel.. collect the mutations afterwards in a new followup github action and report only those as a problem, which were not killed in any of the previous jobs.

alternatively we could run infection on a single php version only